### PR TITLE
[Auth] #434 auth client factory

### LINKED ISSUE #434: Auth client factory

### PR DESCRIPTION
**Pull request by llbartekll**
# Description
- Add Auth Client Factory
- Add empty integration tests for auth

Resolves #434 

## How Has This Been Tested?

integration tests passes

## Due Dilligence

* [ ] Breaking change
* [ ] Requires a documentation update

### CODE REVIEW

File: Sources/Auth/AuthClientFactory.swift [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+public struct AuthClientFactory {
+
+    public static func create(metadata: AppMetadata, account: Account?, relayClient: RelayClient) -> AuthClient {
+        let logger = ConsoleLogger(loggingLevel: .off)
+        let keyValueStorage = UserDefaults.standard
+        let keychainStorage = KeychainStorage(serviceIdentifier: "com.walletconnect.sdk")
+        return AuthClientFactory.create(metadata: metadata, account: account, logger: logger, keyValueStorage: keyValueStorage, keychainStorage: keychainStorage, relayClient: relayClient)
+    }
+
+    static func create(metadata: AppMetadata, account: Account?, logger: ConsoleLogging, keyValueStorage: KeyValueStorage, keychainStorage: KeychainStorageProtocol, relayClient: RelayClient) -> AuthClient {
```
Comment (flypaper0) [inline Sources/Auth/AuthClientFactory.swift:16]: why account is optional? Is it possible to avoid it?

File: Sources/Auth/AuthClientFactory.swift [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        let historyStorage = CodableStore<RPCHistory.Record>(defaults: keyValueStorage, identifier: StorageDomainIdentifiers.jsonRpcHistory.rawValue)
+        let pairingStore = PairingStorage(storage: SequenceStore<WCPairing>(store: .init(defaults: keyValueStorage, identifier: StorageDomainIdentifiers.pairings.rawValue)))
+        let kms = KeyManagementService(keychain: keychainStorage)
+        let serializer = Serializer(kms: kms)
+        let history = RPCHistory(keyValueStore: historyStorage)
+        let networkingInteractor = NetworkingInteractor(relayClient: relayClient, serializer: serializer, rpcHistory: history)
+        let SIWEMessageFormatter = SIWEMessageFormatter()
```
Comment (flypaper0) [inline Sources/Auth/AuthClientFactory.swift:23]: let messageFormatter ?

File: Sources/Auth/AuthClientFactory.swift (hunk 1 above)
Comment (llbartekll) [reply Sources/Auth/AuthClientFactory.swift:16]: yes, App do not need account

**Review by flypaper0 — APPROVED** [review]

File: Sources/Auth/AuthClientFactory.swift (hunk 2 above)
Comment (llbartekll) [reply Sources/Auth/AuthClientFactory.swift]: done

**Review by MisterVants — APPROVED** [review]

**Pull request merged**